config/placeholder_replacement.go

```diff
@@ -153,6 +158,14 @@ func (v *placeholderReplacementVisitor) replacePlaceholders(s string) (string, e
 			}
 			return []byte(replaced)
 		}
+		if environmentPlaceholderRegexp.Match(placeholderKey) {
```

Comment:
could you break out the logic in this file into 2 separate helper functions? It looks messy and kind of hard to follow if you don't know what this doing.

Reply:
Like another layer of nesting? Sure -- I can break it out a bit more -- not sure it will help readability but happy to do it.

Reply:
for example, break out the top function to be like matchOnPackagePlaceholder and this new logic could live in a separate file like matchOnEnvironmentPlaceholder? 

It will help readability because if you just look at those 2 function names its clear what the outer function is doing without having to read / understand a significant amount of code to do so. If you don't know how regex works in depth, this could be hard to follow.

Reply:
I'll try refactoring and see if I can get it looking a bit cleaner -- might try a switch statement.